src/irmin.mli

```diff
@@ -1796,75 +1575,625 @@ module Private: sig
+  (** [repo t] is the repository containing [t]. *)
+
+  val tree: t -> tree Lwt.t
+  (** [tree t] is [t]'s current tree. Contents is not allowed at the
+      root of the tree. *)
+
+  val status: t -> [ `Empty | `Branch of branch | `Commit of commit ]
+  (** [status t] is [t]'s status. It can either be a branch, a commit
+      or empty. *)
+
+  (** Managing the store's heads. *)
+  module Head: sig
+
+    val v: Repo.t -> task -> parents:commit list -> tree -> commit Lwt.t
+    (** [v r task ~parents:p t] is the commit [c] such that:
+        {ul
+        {- [Repo.task_of_commit r c = task]}
+        {- [parents (of_commit r c) = p]}
+        {- [tree (of_commit r c) = t]}}
+    *)
+
+    val list: Repo.t -> commit list Lwt.t
+    (** [list t] is the list of all the heads in local store. Similar
+        to [git rev-list --all]. *)
+
+    val find: t -> commit option Lwt.t
+    (** [find t] is the current head of the store [t]. This works for
+        both persistent and temporary branches. In the case of a
+        persistent branch, this involves getting the the head
+        associated with the branch, so this may block. In the case of
+        a temporary store, it simply returns the current head. Returns
+        [None] if the store has no contents. Similar to [git rev-parse
+        HEAD]. *)
+
+    val get: t -> commit Lwt.t
+    (** Same as {!find} but raise [Invalid_argument] if the store does
+        not have any contents. *)
+
+    val set: t -> commit -> unit Lwt.t
+    (** [set t h] updates [t]'s contents with the contents of the
+        commit [h]. Can cause data loss as it discards the current
+        contents. Similar to [git reset --hard <hash>]. *)
+
+    val fast_forward: t -> ?max_depth:int -> ?n:int -> commit -> bool Lwt.t
+    (** [fast_forward t h] is similar to {!update} but the [t]'s head
+        is updated to [h] only if [h] is stricly in the future of
+        [t]'s current head. Return [false] if it is not the case. If
+        present, [max_depth] or [n] are used to limit the search space
+        of the lowest common ancestors (see {!lcas}). *)
+
+    val test_and_set:
+      t -> test:commit option -> set:commit option -> bool Lwt.t
+    (** Same as {!update_head} but check that the value is [test] before
+        updating to [set]. Use {!update} or {!merge} instead if
+        possible. *)
+
+    (** [merge ~into:t ?max_head ?n commit] merges the contents of the
+        commit associated to [commit] into [t]. [max_depth] is the
+        maximal depth used for getting the lowest common ancestor. [n]
+        is the maximum number of lowest common ancestors. If present,
+        [max_depth] or [n] are used to limit the search space of the
+        lowest common ancestors (see {!lcas}). *)
+    val merge: into:t -> task -> ?max_depth:int -> ?n:int -> commit ->
+      (unit, Merge.conflict) result Lwt.t
+
+    val parents: t -> commit list Lwt.t
+    (** [parents t] are [t]'s parent commits.  *)
+
+  end
```

Comment:
Generally, this API seems a bit confused about whether `Store.Head` is about stores (mutable refs) or commits (immutable values).